### PR TITLE
closes #657

### LINKED ISSUE #657: f != originalToNewFn.end()
I have a library which crashes when using Enzyme. After the IO updates it now failes at the following assertion:
`f != originalToNewFn.end()`

I have the errormessage / stacktrace here
[err.txt](https://github.com/EnzymeAD/Enzyme/files/8712129/err.txt)

Is there anything obviously wrong around this assertion?
I will work on providing a minimized src code example.

Also, how would I pass the extra enzyme-print flags, when using LLDEnzyme?

### PR DESCRIPTION
**Pull request by ZuseZ4**
@wsmoses let me know if I took your comment too literal. Seems to fix at least my Bug.

### CODE REVIEW

File: enzyme/Enzyme/GradientUtils.cpp [hunk 1]
```diff
@@ -3821,6 +3821,10 @@ Value *GradientUtils::invertPointerM(Value *const oval, IRBuilder<> &BuilderM,
       }
     }
 
+    if (auto arg = dyn_cast<ConstantExpr>(oval)) {
+      return oval;
```
Comment (tgymnich) [inline enzyme/Enzyme/GradientUtils.cpp:3825]: You might also want to handle the vector mode case here. E.g. wrap this inside of applyChainRule.

Comment (tgymnich) [inline enzyme/Enzyme/GradientUtils.cpp:3825]: You might also want to handle the vector mode case here. E.g. wrap this inside of applyChainRule.

**Review by wsmoses — APPROVED** [review]
LGTM, but address Tim's comment and please add a test.

File: enzyme/Enzyme/GradientUtils.cpp (hunk 1 above)
Comment (ZuseZ4) [reply enzyme/Enzyme/GradientUtils.cpp]: done.

File: enzyme/Enzyme/GradientUtils.cpp [hunk 2]
```diff
@@ -3821,6 +3821,11 @@ Value *GradientUtils::invertPointerM(Value *const oval, IRBuilder<> &BuilderM,
       }
     }
 
+    if (auto arg = dyn_cast<ConstantExpr>(oval)) {
+      auto rule = [&oval]() { return Constant::getNullValue(oval->getType()); };
```
Comment (wsmoses) [inline enzyme/Enzyme/GradientUtils.cpp:3825]: Oh you don't want to return the null, here but arg (null case is already taken care of above.

Comment (ZuseZ4) [reply enzyme/Enzyme/GradientUtils.cpp:3825]: Thanks, I was already wondering about the test-case changes.

**Comment by ZuseZ4** [comment]
@wsmoses Do you mind if I revert the test for now?
It not only triggers this, but also a second bug, I don't have any example which just triggers the first bug.
Once the second bug is resolved, I can add the testcase, without causing CI failures.

**Comment by wsmoses** [comment]
I'm fine with that, but open an issue for the new bug, and reference this PR to make sure we remember that the test for it should also include the test for this.

**Pull request merged**